Firestore/core/test/unit/remote/serializer_test.cc

```diff
@@ -1512,6 +1513,62 @@ TEST_F(SerializerTest, EncodesResumeTokens) {
   ExpectRoundTrip(model, proto);
 }
 
+TEST_F(SerializerTest, EncodesExpectedCount) {
+  core::Query q = Query("docs");
+  TargetData model(q.ToTarget(), 1, 0, QueryPurpose::Listen,
+                   SnapshotVersion::None(), SnapshotVersion::None(),
+                   Bytes({1, 2, 3}), /*expected_count=*/2);
+
+  v1::Target proto;
+  proto.mutable_query()->set_parent(ResourceName(""));
+  proto.set_target_id(1);
+
+  v1::StructuredQuery::CollectionSelector from;
+  from.set_collection_id("docs");
+  *proto.mutable_query()->mutable_structured_query()->add_from() =
+      std::move(from);
+
+  v1::StructuredQuery::Order order;
+  order.mutable_field()->set_field_path(FieldPath::kDocumentKeyPath);
+  order.set_direction(v1::StructuredQuery::ASCENDING);
+  *proto.mutable_query()->mutable_structured_query()->add_order_by() =
+      std::move(order);
+
+  proto.set_resume_token("\001\002\003");
+
+  google::protobuf::Int32Value* expectedCount = new Int32Value();
```

Comment:
Instead of `new Int32Value()` use `Inte32Value::New()`. I'm not 100% sure what the difference is, but the only other place in our code base that calls one of the `set_allocated_XXX()` functions uses this idiom: https://github.com/firebase/firebase-ios-sdk/blob/172c7520b3b6b2f66926e00263d3ec7af6d3393f/Firestore/core/test/unit/remote/serializer_test.cc#L1836-L1837